### PR TITLE
Fix segmentation fault in tf.image.crop_and_resize when boxes is inf or nan

### LINKED ISSUE #42129: segfault in `tf.image.crop_and_resize` when `boxes` contains large value
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.2.0-rc4-8-g2b96f3662b 2.2.0
- Python version: 3.6.9
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
`tf.image.crop_and_resize` segfault when there is a very large value in `boxes`. Can also be reproduced in nightly version

**Describe the expected behavior**
Expect no segfault
**Standalone code to reproduce the issue**

~~~python
import tensorflow as tf
tf.image.crop_and_resize(image=tf.zeros((2,1,1,1)), boxes=[[1.0e+40, 0,0,0]], box_indices=[1], crop_size=[1,1])
~~~

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

~~~python
Segmentation fault (core dumped)
~~~

### PR DESCRIPTION
**Pull request by yongtang**
This fix tries to address the issue raised in #42129 where segmentation fault
happened in tf.image.crop_and_resize when boxes is inf or nan.

This fix adds the check to make sure boxes is not inf or nan (isfinite)

This fix fixes #42129.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tensorflow/core/kernels/crop_and_resize_op.cc [hunk 1]
```diff
@@ -256,6 +268,7 @@ struct CropAndResize<CPUDevice, T> {
             continue;
           }
           if (method_name == "bilinear") {
+
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/crop_and_resize_op.cc:271]:
```suggestion
```

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by gbaned** [comment]
@yongtang Can you please resolve conflicts? Thanks!

**Comment by gbaned** [comment]
@yongtang, Any update on this PR? Please. Thanks!

**Comment by yongtang** [comment]
Due to the way GPU/CPU pipeline works in the kernel, it looks like to do a nan check on GPU is not exactly straightforward. That was causing the GPU test failure.

Let me take another look and see how to resolve this issue.

File: tensorflow/core/kernels/image/crop_and_resize_op.cc [hunk 2]
```diff
@@ -256,6 +260,7 @@ struct CropAndResize<CPUDevice, T> {
             continue;
           }
           if (method_name == "bilinear") {
+
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/image/crop_and_resize_op.cc:263]:
```suggestion
```

**Review by mihaimaruseac — DISMISSED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Review by sanjoy — CHANGES_REQUESTED** [review]
This PR is failing some tests.  Are you sure that the `boxes.tensor<float, 2>().isfinite().all()` line is correct even when the operation is running on a GPU?

**Comment by gbaned** [comment]
@yongtang  Can you please check @sanjoy's comments and keep us posted ? Thanks!

**Comment by mihaimaruseac** [comment]
The test added by the PR does not fail in CUDA if I remove the C++ check that causes the segfaults

**Comment by mihaimaruseac** [comment]
I think I got this fully fixed internally. Running a full suite of tests and sending for review tomorrow.

**Pull request merged**